### ISSUE TITLE
[GR-57946] Implement JS String Builtins Proposal for WebAssembly

### ISSUE THREAD

**Issue by fniephaus** [oracle/graal#9766]
### TL;DR

We plan to implement [the JS String Builtins proposal](https://github.com/WebAssembly/js-string-builtins) in GraalWasm.

### Details

The JS String Builtins proposal makes it easier to call into the methods of JavaScript's String objects. Having a low-overhead standardized way of accessing JS Strings from WebAssembly means that languages can use JS Strings when compiling to WebAssembly. Languages that take advantage of this new feature will be able to compile programs that use strings into smaller modules. They will also benefit from the efficient TruffleString implementation that is used in Graal Languages like GraalJS.

More details at https://github.com/WebAssembly/js-string-builtins/blob/main/proposals/js-string-builtins/Overview.md
